src/components/auth/Auth.jsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import { useLoginMutation, useRegisterMutation } from "./authSlice";
+import { useNavigate } from "react-router-dom";
+import "../../styles/Auth.css";
+
+function Auth() {
+  const navigate = useNavigate();
+
+  //swap between register and login
+  const [isLogin, setIsLogin] = useState(true);
+  const authAction = isLogin ? "Login" : "Register";
+  //register or login message
+  const altCopy = isLogin
+    ? "Register for an account here"
+    : "Already registered? Login here";
+
+  //mutate data for login/registration
+  const [login, { error: loginError }] = useLoginMutation();
+  const [register, { error: registerError }] = useRegisterMutation();
+
+  //email and password to state
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [firstName, setFirstName] = useState("");
+
+  //send user email/password to api
+  const attemptAuth = async (event) => {
+    event.preventDefault();
+
+    //login or register
+    const loginMethod = isLogin ? login : register;
+    const credentials = { email, password, firstName };
```

Comment:
Since the schema uses "firstname" notation, can we update this to "firstname" as well so the model recognizes the parameter?